### ISSUE TITLE
Wallet Error in Ubuntu

### ISSUE THREAD

**Issue by nikolayg** [PIVX-Project/PIVX#87]
I'd like to report a bug with the wallet. It just failed out of the blue with:

```
pivx-qt: malloc.c:2392: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
Aborted (core dumped)
```
I don't know your code, but I'd suggest you refactor that mega expression  ...

My wallet is now stuck synchronising at `3 hours behind` and does not progress ...

**Comment by grigio** [comment]
I've a similar crash after several hour it is up
```
pivx-qt: malloc.c:2395: sysmalloc: Assertion `(old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)' failed.
[1]    6355 abort      ./pivx-qt
```

**Comment by presstab** [comment]
@nikolayg malloc is part of the C library, not something written by us. 

Most likely that this would be some type of memory corruption fault that will require further debugging to pinpoint.

**Comment by Fuzzbawls** [comment]
As @presstab said, this could be caused by issues outside the wallet's code. Unless further details can be provided, we will need to close the issue.

**Comment by presstab** [comment]
I have had this issue pop up for me a few times as well. I do think that it is caused by Pivx, only just that the assertion occurs within the C library. I have only noticed this in QT wallet. I will run a fault detection tool such as cppcheck and see if anything is flagged.

**Comment by 4ml** [comment]
I can report a similar issue. I'm running pivx-2.2.0 on Ubuntu Server 14.04 LTS 64bit. My VPS has 1 GB RAM and 20 GB HD.

I'm happy to share my debug.log if someone can confirm I won't be making myself vulnerable by doing so.

**Comment by abderem** [comment]
Nothing to do with Ubuntu but it seems with all Linux distros, see similar report #185 
Please fix this issue, as I rent a server for this and haven't had a stake in weeks...

**Comment by Fuzzbawls** [comment]
fixed in #206
